src/data/polynomial/degree/definitions.lean

```diff
@@ -835,6 +843,84 @@ lemma le_nat_degree_of_coe_le_degree (hdeg : ↑n ≤ p.degree) :
   n ≤ p.nat_degree :=
 with_bot.coe_le_coe.mp ((degree_eq_nat_degree $ ne_zero_of_coe_le_degree hdeg) ▸ hdeg)
 
+lemma degree_sum_eq_of_disjoint (f : S → polynomial R) (s : finset S)
+  (h : set.pairwise ↑(s.filter (λ i, f i ≠ 0)) (ne on (degree ∘ f))) :
+  degree (∑ i in s, f i) = s.sup (λ i, degree (f i)) :=
+begin
+  classical,
```

Comment:
Minor comment. These proofs are somewhat longish, so it feels a bit odd to have these lemmas in a file called `definitions.lean`. Does it make sense to move them to `lemmas.lean`?